tests/eas/generic.py

```diff
@@ -0,0 +1,433 @@
+        """
+        Get a DataFrame mapping task names to the CPU they ran on
+
+        Use the sched_switch trace event to find which CPU each task ran
+        on. Does not reflect idleness - tasks not running are shown as running
+        on the last CPU they woke ran on.
+
+        :param experiment: The :class:Experiment to examine
+        :returns: A Pandas DataFrame with a column for each task, showing the
+                  CPU that the task was "on" at each moment in time
+        """
+        tasks = experiment.wload.tasks.keys()
+        trace = self.get_trace(experiment)
+
+        df = trace.ftrace.sched_switch.data_frame[['next_comm', '__cpu']]
+        df = df[df['next_comm'].isin(tasks)]
+        df = df.pivot(index=df.index, columns='next_comm').fillna(method='ffill')
+        cpu_df = df['__cpu']
+        # Drop consecutive duplicates
+        cpu_df = cpu_df[(cpu_df.shift(+1) != cpu_df).any(axis=1)]
+        return cpu_df
+
+    def get_power_df(self, experiment):
+        """
+        Considering only the task placement, estimate power usage over time
+
+        Examine a trace and use :meth:EnergyModel.estimate_from_cpu_util to get
+        a DataFrame showing the estimated power usage over time. This assumes
+        perfect cpuidle and cpufreq behaviour.
+
+        :param experiment: The :class:Experiment to examine
+        :returns: A Pandas DataFrame with a column each node in the energy model
+                  (keyed with a tuple of the CPUs contained by that node) and a
+                  "power" column with the sum of other columns. Shows the
+                  estimated power over time.
+        """
+        task_cpu_df = self.get_task_cpu_df(experiment)
+        task_utils_df = self.get_task_utils_df(experiment)
+
+        tasks = experiment.wload.tasks.keys()
+
+        # Create a combined DataFrame with the utilization of a task and the CPU
+        # it was running on at each moment. Looks like:
+        #                       utils                  cpus
+        #          task_wmig0 task_wmig1 task_wmig0 task_wmig1
+        # 2.375056      102.4      102.4        NaN        NaN
+        # 2.375105      102.4      102.4        2.0        NaN
+
+        df = pd.concat([task_utils_df, task_cpu_df],
+                       axis=1, keys=['utils', 'cpus'])
+        df = df.sort_index().fillna(method='ffill')
+        nrg_model = self.executor.te.nrg_model
+
+        # Now make a DataFrame with the estimated power at each moment.
+        def est_power(row):
+            cpu_utils = [0 for cpu in nrg_model.cpus]
+            for task in tasks:
+                cpu = row['cpus'][task]
+                util = row['utils'][task]
+                if not isnan(cpu):
+                    cpu_utils[int(cpu)] += util
+            power = nrg_model.estimate_from_cpu_util(cpu_utils)
+            columns = power.keys()
+            return pd.Series([power[c] for c in columns], index=columns)
```

Comment:
Yep. I think I'll remove the "power" column and sort the node columns by depth-first order.